flow/connectors/clickhouse/normalize.go

```diff
@@ -321,23 +326,43 @@ func (c *ClickHouseConnector) NormalizeRecords(
 					colName,
 					dstColName,
 				))
+				projectionUpdate.WriteString(fmt.Sprintf(
+					"toDate32(parseDateTime64BestEffortOrNull(JSONExtractString(_peerdb_match_data, '%s'))) AS `%s`,",
+					colName,
+					dstColName,
+				))
 			case "DateTime64(6)", "Nullable(DateTime64(6))":
 				projection.WriteString(fmt.Sprintf(
 					"parseDateTime64BestEffortOrNull(JSONExtractString(_peerdb_data, '%s')) AS `%s`,",
 					colName,
 					dstColName,
 				))
+				projectionUpdate.WriteString(fmt.Sprintf(
+					"parseDateTime64BestEffortOrNull(JSONExtractString(_peerdb_match_data, '%s')) AS `%s`,",
+					colName,
+					dstColName,
+				))
 			default:
 				projection.WriteString(fmt.Sprintf("JSONExtract(_peerdb_data, '%s', '%s') AS `%s`,", colName, clickHouseType, dstColName))
+				projectionUpdate.WriteString(fmt.Sprintf(
+					"JSONExtract(_peerdb_match_data, '%s', '%s') AS `%s`,",
+					colName,
+					clickHouseType,
+					dstColName,
+				))
 			}
 		}
 
 		// add _peerdb_sign as _peerdb_record_type / 2
 		projection.WriteString(fmt.Sprintf("intDiv(_peerdb_record_type, 2) AS `%s`,", signColName))
+		// projectionUpdate generates delete on previous record, so _peerdb_record_type is filled in as 2
+		projectionUpdate.WriteString(fmt.Sprintf("1 AS `%s`,", signColName))
 		colSelector.WriteString(fmt.Sprintf("`%s`,", signColName))
 
 		// add _peerdb_timestamp as _peerdb_version
 		projection.WriteString(fmt.Sprintf("_peerdb_timestamp AS `%s`", versionColName))
+		// decrement timestamp by 1 so delete is ordered before update's new data
+		projectionUpdate.WriteString(fmt.Sprintf("_peerdb_timestamp - 1 AS `%s`", versionColName))
```

Comment:
in theory this could be weird if two updates happened after each other by 1 tick with the second update on the primary key